OmarManassra/task_manager.py

```diff
@@ -0,0 +1,87 @@
+from  utils import validate_date,get_valid_input,validate_status,validate_priority
+from task import Task,UrgentTask
+# Dictionory contains all the tasks 
+tasks_dict = {}
+def add_task():
+    """ Call the Exception Handling Functions in the other module at every step it should be called"""
+    title=input("Enter the title")
+    description=input("enter the description")
+    due_date = input("Enter due date (YYYY-MM-DD): ")
+    validate_date(due_date)
+    status = input("Enter status ('InProgress' or 'Completed'): ")
+    validate_status(status)
+    try:
+        """ Check if its Urgant """
+        is_urgant=input("enter if urgant or not (True or False)")
+        if is_urgant:
+            priority = input("Enter priority ('High', 'Medium', 'Low'): ")
+            validate_priority(priority)
+            addUrgent= UrgentTask(priority,title,description,due_date,status)
+            tasks_dict[addUrgent.task_id]=addUrgent
+        else:
+            addtask=Task(title,description,due_date,status)
+            tasks_dict[addtask.task_id]=addtask
+            """ Check if there any Exception in adding tasks to the task list"""
+        
+    except ValueError as e:
+        print (e)
+
+def view_tasks():
+    """ Display the tasks"""
+    if tasks_dict:
+         for task in tasks_dict.values():
+            task.display()
+    else:
+        print ("No tasks added")
+
+
+def update_task():
```

Comment:
What happens if the `task_id `is not in the `tasks_dict`? 
I think that the task manager should tell the user that the `id` does not exist before entering all this information